backends/llamacpp/src/main.rs

```diff
@@ -205,24 +206,51 @@ async fn main() -> Result<(), RouterError> {
             token,
             ..Default::default()
         };
-        Tokenizer::from_pretrained(args.model_id.clone(), Some(params))?
+        Tokenizer::from_pretrained(&args.model_id, Some(params))?
+    };
+
+    let model_gguf = match args.model_gguf {
+        Some(model_gguf) => model_gguf,
+        None => {
+            let make_gguf = match std::env::var("MAKE_GGUF") {
+                Ok(make_gguf) => make_gguf,
+                Err(e) => {
+                    error!("Missing env: MAKE_GGUF");
+                    return Err(RouterError::VarError(e));
+                }
+            };
+            let model_gguf = "models/model.gguf".to_string();
+
+            let status = Command::new(make_gguf)
+                .arg(&model_gguf)
+                .arg(&args.model_id)
+                .arg(&args.revision)
+                .spawn()?
+                .wait()
+                .await?;
+
+            if !status.success() {
+                error!("Failed to generate GGUF");
+            }
+            model_gguf
+        }
```

Comment:
Do we want this to be handled by the server? Or should it be handled by an entrypoint in the container?

Reply:
I initially implemented it in an entrypoint, but I had to manage arguments and environment variables and mix it with users choices or doing something like the launcher...